backend/tests/test_models_endpoints.py

```diff
@@ -1,24 +1,216 @@
+    assert response.json() == {
+        "initialConcentrations": {},
+        "finalConcentrations": {},
+        "panels": [],
+    }
+
+
+@pytest.mark.parametrize(
+    "valid_substances,concs,expected_concs",
+    [(["H2"], {"H2": 10}, None), ([], {"H2": 10}, {}), (["H2"], {}, {"H2": 0.0})],
+)
+def test_dummy_model_only_valid_substances_are_present(
+    client, dummy_model, monkeypatch, valid_substances, concs, expected_concs
+):
+    monkeypatch.setattr(dummy_model, "valid_substances", valid_substances)
+
+    response = client.post(
+        f"/models/{dummy_model.model_id}/runs",
+        json={
+            "concs": concs,
+            "settings": {},
+        },
+    )
+    response.raise_for_status()
+    assert response.json() == {
+        "initialConcentrations": concs if expected_concs is None else expected_concs,
+        "finalConcentrations": concs if expected_concs is None else expected_concs,
+        "panels": [],
+    }
+
+
+class _DummyModelParameters(BaseParameters):
+    some_field: int = Parameter(0)
+
+
+class _DummyModelParametersMinMax(BaseParameters):
+    some_field: int = Parameter(0, min=-5, max=5)
+
+
+def test_dummy_has_correct_parameter_name(client, monkeypatch, dummy_model):
+    async def run(self):
+        return self.concentrations, JsonResult(data=self.parameters)
+
+    monkeypatch.setitem(
+        dummy_model.__annotations__, "parameters", _DummyModelParameters
+    )
+    monkeypatch.setattr(dummy_model, "run", run)
+
+    response = client.get_json("/models")
+
+    assert response == [
+        {
+            "accessError": None,
+            "displayName": "Dummy Model",
+            "modelId": "dummy",
+            "parameters": {
+                # Notice that it's "someField" and not "some_field"
+                "someField": {
+                    "__type__": "AcidwatchParameter",
+                    "choices": None,
+                    "convertibleUnit": None,
+                    "default": 0,
+                    "description": None,
+                    "label": None,
+                    "title": "Somefield",
+                    "type": "integer",
+                    "unit": None,
+                }
+            },
+            "validSubstances": [],
+        }
+    ]
+
+
+@pytest.mark.parametrize(
+    "parameters_class,input_parameters,expected",
+    [
+        pytest.param(
+            None,
+            {"someField": 123},
+            "Model dummy expected no parameters, got {'someField': 123}",
+            id="no params defined",
+        ),
+        pytest.param(
+            _DummyModelParameters,
+            {"someField": 123},
+            {"someField": 123},
+            id="valid camel case param",
+        ),
+        pytest.param(_DummyModelParameters, {}, {"someField": 0}, id="default value"),
+        pytest.param(
+            _DummyModelParameters,
+            {"some_field": 123},
+            {"someField": 123},
+            id="valid snake case param",
```

Comment:
so we pretty much just handle both ways of stating parameter names in our api?

Reply:
Yeah apparently Pydantic does both. I figure there's no point in changing it, but it's good to have tests that check in case the behaviour changes. Note that the frontend will always use the camelCase version, but someone using the API directly might end up using snake_case.

Reply:
Hopefully this isn't something that could potentially bite us. I think we should be good.. "famous last words"